### ISSUE TITLE
HLS streaming .ts segment error stop video

### ISSUE THREAD

**Issue by onigetoc** [videojs/videojs-contrib-hls#1063]
HLS streaming .ts segment error stop video.

Sometime, if i look at the console.log NETWORK (Google chrome inspector)
Method: GET
Status: Canceled
There's segment error and the video stop and never restart sometime i should manually pause de video et click play again.

I do not know if it's have to do with my internet bandwith, internet speed, or m3u8 video server.
but for me it,s happen often with this video
Infowars: http://infowarslive-lh.akamaihd.net/i/infowarslivestream_1@353459/index_800_av-p.m3u8

If it stop, i click pause then play to restart it. but it may restart by itself but may wait to the next working .ts segment.

**Comment by gesinger** [comment]
Hey @onigetoc , I left the stream running for a bit, but I didn't see any errors. Do you happen to have any other information around the request that might help? And what version of videojs-contrib-hls are you using? Thanks.

**Comment by onigetoc** [comment]
It's hard to tell, it may be my internet connection, but it do not stop on my iPad and Android tablet but it stop on Chrome on desktop.
The also do not stop with this m3u8 chrome extension hls player (Sometime, i can see the segment error but it do not give up and the .ts segment is loading)
https://chrome.google.com/webstore/detail/native-hls-playback/emnphkkblegpebimobpbekeedfgemhof
(Created with the original hls.js plugin)

On video js, if the .ts segment fail it do not try to load it like the Chrome extension.

Many m3u8 videos stops, it's hard to tell why some stop and some not. But they never reload or restart, they also stop on Chromecast. (But like i said, they do not stop on my two tablets (IOS / Android))

-----------------------------------
Just another note with no relation the above problem:
Videos who are blocked in the header (403 or 404) are not blocked on mobile tablet.

**Comment by gesinger** [comment]
@onigetoc , how long do you normally run that stream before it stops?

**Comment by onigetoc** [comment]
May be 2 to 3 minutes, tryed again on Android and it work but stop on desktop Chrome

**Comment by gesinger** [comment]
@onigetoc , I left it running for over 10 minutes, often in a background tab, but it never stopped playing for me. Is there anything in the console or any more network trace info?

**Comment by onigetoc** [comment]
Console log after around 2 min give me a .ts file cancel 403.
after it seem to try to load the .m3u8 file but it's never followed by a .ts file.

It is my internet connection, or the external .ts file that was down some seconds and it do not try to reload this file?
Because, most of the m3u8 video work well and do not fail loading the .ts files.

**Comment by gesinger** [comment]
@onigetoc , that sounds like it has to do with the server if it is returning a 403 or 404. However, we do have logic in the latest version of videojs-contrib-hls (v5.4.0) that will not blacklist the final rendition on a 403/404 and will continue trying. This means that even if there is a momentary loss of connection or a server error, the player won't give up, and will retry after a delay. Would you be able to give that version a shot and see if you still experience the error?

**Comment by onigetoc** [comment]
Hi, i test from CDN but the 5.4 is not there. i guess i should wait a little bit.
OK
https://cdnjs.cloudflare.com/ajax/libs/videojs-contrib-hls/5.3.3/videojs-contrib-hls.min.js
Not OK
https://cdnjs.cloudflare.com/ajax/libs/videojs-contrib-hls/5.4/videojs-contrib-hls.min.js

**Comment by gesinger** [comment]
Hey @onigetoc , you may have to include the full 5.4.0 in the URL: https://cdnjs.cloudflare.com/ajax/libs/videojs-contrib-hls/5.4.0/videojs-contrib-hls.min.js

**Comment by onigetoc** [comment]
It may depend of the time of the day i'm checking the video, but until now i got no connection error but the video stopped again after around 3 minutes.
Now it try again and again to load this URL: 
http://infowarslive-lh.akamaihd.net/i/infowarslivestream_1@353459/index_800_av-p.m3u8
But not followed by a .ts file.

**Comment by onigetoc** [comment]
And every time a load my web page i got this error in the begining.

> VIDEOJS: WARN: Problem encountered with the current HLS playlist. Trying again since it is the final playlist.
> c.logByType @ video.min.js:18
> i.warn @ video.min.js:18
> value @ videojs-contrib-hls.min.js:1
> (anonymous) @ videojs-contrib-hls.min.js:1
> d.dispatcher.d.dispatcher @ video.min.js:18
> k @ video.min.js:18
> g.trigger @ video.min.js:14
> value @ videojs-contrib-hls.min.js:2
> (anonymous) @ videojs-contrib-hls.min.js:1
> (anonymous) @ videojs-contrib-hls.min.js:1
> (anonymous) @ videojs-contrib-hls.min.js:3
> j @ video.min.js:19
> f @ video.min.js:19
> b @ video.min.js:19

**Comment by onigetoc** [comment]
Now the sound is not in sync with the video

**Comment by gesinger** [comment]
These errors sound like they are coming from the stream itself. There isn't much we can do for the server responses, since the player seems to be making valid segment requests.

**Comment by onigetoc** [comment]
Now the hls plugin try too much to find inexisting or not working video, it never stop trying loading a failing video.
http://streaming-hub.com/stream/arte.m3u8

I think these video block call from other domain, but the 5.4 hls plugin never stop trying to load it.

**Comment by gesinger** [comment]
That's correct @onigetoc . One of the recent changes to the player is that we don't blacklist the final rendition, meaning that the player will keep trying that stream. This is normally fine, as the stream should be able to play within a player without trouble, and normally when errors happen it is due to the server being down temporarily, rather than a complete block due to permissions (in which case the stream isn't usually meant for that domain/user to play). However, it would be nice for developers to be able to react to such an event, so I opened: https://github.com/videojs/videojs-contrib-hls/issues/1074

For the purpose of this issue though, it looks like the server was creating the errors, rather than the player itself. Thanks for all of the examples and info. Closing this ticket, as the player is working as designed.
